content/articles/boosting-algorithms-part-1/index.md

```diff
@@ -0,0 +1,251 @@
+
+![Boosting](/engineering-education/boosting-algorithms-part-1/boosting.png)
+
+*Source: [Towards Data Science](https://towardsdatascience.com/ensemble-methods-bagging-boosting-and-stacking-c9214a10a205)*
+
+### Data Preprocessing
+The article uses the [UCI Machine Learning Mushroom Dataset](https://www.kaggle.com/uciml/mushroom-classification) to implement the AdaBoost and XGBoost algorithms. For the set of features in the dataset, the task is to identify whether the type of mushroom is poisonous or edible. The reader is encouraged to download the dataset and follow along with the code blocks in the article for better understanding.
+
+First, let's import the required libraries.
+
+```py
+import numpy as np
+import pandas as pd
+from sklearn.ensemble import AdaBoostClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import LabelEncoder
+from sklearn.preprocessing import StandardScaler
+```
+
+Now, let's import the dataset using the `read_csv()` method in Pandas and analyze the number of distinct categories in each feature. If a feature has only one unique value, we can drop it, as it has no significance while building the model.
+
+```py
+df = pd.read_csv("mushrooms.csv")
+for col in df.columns:
+    print('Number of unique values in',col,'is',len(df[col].unique()))
+```
+**Output**
+
+```bash
+Number of unique values in class is 2
+Number of unique values in cap-shape is 6
+Number of unique values in cap-surface is 4
+Number of unique values in cap-color is 10
+Number of unique values in bruises is 2
+Number of unique values in odor is 9
+Number of unique values in gill-attachment is 2
+Number of unique values in gill-spacing is 2
+Number of unique values in gill-size is 2
+Number of unique values in gill-color is 12
+Number of unique values in stalk-shape is 2
+Number of unique values in stalk-root is 5
+Number of unique values in stalk-surface-above-ring is 4
+Number of unique values in stalk-surface-below-ring is 4
+Number of unique values in stalk-color-above-ring is 9
+Number of unique values in stalk-color-below-ring is 9
+Number of unique values in veil-type is 1
+Number of unique values in veil-color is 4
+Number of unique values in ring-number is 3
+Number of unique values in ring-type is 5
+Number of unique values in spore-print-color is 9
+Number of unique values in population is 6
+Number of unique values in habitat is 7
+```
+
+As you can see `Number of unique values in veil-type is 1`, the feature `veil-type` has only one distinct value in it and hence, can be dropped.
+
+```py
+df = df.drop("veil-type",axis=1)
+```
+
+Since machine learning models prefer numerical data, let us convert the dataset to numbers by encoding it.
+
+```py
+label_encoder = LabelEncoder()
```

Comment:
Give 1 - 2 lines of explanation of how `LabelEncoder()` works. And, add any external link that the reader might find it useful to understand more about label encoding.